src/transformers/models/auto/tokenization_auto.py

```diff
@@ -895,9 +895,12 @@ def from_pretrained(cls, pretrained_model_name_or_path, *inputs, **kwargs):
             )
         elif config_tokenizer_class is not None:
             tokenizer_class = None
+            class_name = config_tokenizer_class.rstrip("Tokenizer").lower()
```

Comment:
this is probably a bit brittle no? `config_tokenizer_class` can be `LayoutLVM` but the id of the model is `layout_lvm` so not juste lowered!

Reply:
```python
#slow_class_name is something like SiglipTokenizer
def tokenizer_fast_class_from_name(slow_class_name: str):
    for module_name, tokenizers in TOKENIZER_MAPPING_NAMES.items():
        # search the VALUES of the dict to find the slow class
        if class_name in tokenizers:
            # check if the slow_class + "Fast" is in the values
            if f"{class_name}Fast" in tokenizers:
                module_name = model_type_to_module_name( f"{class_name}Fast")
                module = importlib.import_module(f".{module_name}", "transformers.models")
                try:
                    return getattr(module, class_name)
                except AttributeError:
                    continue
            # check if "PreTrainedTokenizerFast" is in the values
            elif "PreTrainedTokenizerFast" in tokenizers:
                return PreTrainedTokenizerFast
            else:
                return None
          
```